pkg/apis/apps.kubermatic/v1/application_definition.go

```diff
@@ -151,8 +151,12 @@ type ApplicationTemplate struct {
 }
 
 type ApplicationVersion struct {
-	// Version of the application (eg v1.2.3)
-	Version string `json:"version"`
+	// +kubebuilder:validation:Pattern:=v?([0-9]+)(\.[0-9]+)?(\.[0-9]+)?(-([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?(\+([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?
```

Comment:
I really don't think we should copy/replicate some Go package's internal regex to parse version numbers. The same effect (rejecting invalid resources) can be achieved with webhooks as well.

Reply:
yeah that is true. Since I'll have to touch the webhook anyways I am going to have a look. 

Initial idea was to make it clearer to the user that a valid semVer should be entered here. The other part is, it is essentially using the same regex as the masterminds/semver

But I'll have a look

Reply:
That is true, but I would consider the regex to be semver's implementation detail. What if at some point they update their parsing logic to not follow a regex, but be based on string splitting and stuff? Then we're suddenly left in the dark. A nice doc comment ("This must be a valid semver version, optionally with a leading 'v' prefix.") is much more helpful than presenting a long regex to the developer ^^